### PR TITLE
QE-159

### PR DESCRIPTION
**Pull request by etaylork**
**Objective:**
Create a generalized workflow that any repo within hinge-health can leverage in order to perform static code analysis and code coverage scans through SonarQube.

This workflow will do as follows:
1. run the set of unit tests depending on the underlying coding language of the repo
2. execute the SonarQube scanner

### CODE REVIEW

File: .github/workflows/static-code-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      sonar_token:
+        description: 'Access token to interact with SonarQube instance.'
+        required: true
+      shared_packages_read_token:
+        description: 'Token to download the required shared node packages for compilation.'
+        required: false
+
+jobs:
+  nodejs_test:
+    name: Unit Tests - NodeJs
```
Comment (etaylork) [inline .github/workflows/static-code-analysis.yml:28]: @aliwatters question for this as maybe we want to rethink the design of this github action to be less defined and more abstract. Would it be better to remove the steps that involve downloading the required packages, dependencies, etc.. to run the unit tests and just have a step that executes the command that runs the unit tests and outputs the coverage reports ? 

The pods themselves will have to be sure to download the required packages and such to run the tests/output the coverage reports within the GH action but this gives them the freedom to add in this GH action workflow call within any workflow they choose whether it's a separate workflow just for static code analysis or one of their existing workflows.

File: .github/workflows/static-code-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+name: SonarQube Static Code Analysis
+
+on:
+  workflow_call:
+    inputs:
+      sonar_host_url:
```
[the remark this answers is not in the document]
Comment (aliwatters) [reply .github/workflows/static-code-analysis.yml:6]: nice -- we can set that up as an organization secret in GH -- set once and forget (well... update rarely)

File: .github/workflows/static-code-analysis.yml (hunk 1 above)
Comment (aliwatters) [reply .github/workflows/static-code-analysis.yml:28]: yes -- I was thinking along these lines.

Eg. `with: test-command-that-outputs coverage/locov`

File: .github/workflows/static-code-analysis.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (aliwatters) [reply .github/workflows/static-code-analysis.yml:6]: `sonary_host_url: ${{ secrets.SONAR__HOST_URL }}`
 
example -- I can add/update that host

**Pull request closed without merging**